[release/5.0-rc2] Add public API baselines for HTML and HTTP

dougbu opened the pull request:
- part of #24347

nits: take a few VS suggestions in opened files
- e.g. use pattern matching and `using` statements a bit more

dougbu committed:
[release/5.0-rc2] Add public API baselines for HTML and HTTP
- part of #24347

nits: take a few VS suggestions in opened files
- e.g. use pattern matching and `using` statements a bit more

dougbu commented on src/Html/Abstractions/src/HtmlContentBuilder.cs:
```diff
@@ -183,8 +183,7 @@ public void WriteTo(TextWriter writer, HtmlEncoder encoder)
 
-                var entryAsString = entry as string;
-                if (entryAsString != null)
+                if (entry is string entryAsString)
```
Reviewers: I recommend mainly checking that taking VS suggestions couldn't break anything. The build will validate the new `*.txt` files.

dougbu commented on src/Http/Routing.Abstractions/src/Properties/AssemblyInfo.cs:
```diff
@@ -9,7 +9,10 @@
+// This warning cannot be suppressed in the Enumerator source because constructor is implicit.
+#pragma warning disable RS0016 // Symbol 'implicit constructor for 'Enumerator'' is not part of the declared API.
 [assembly: TypeForwardedTo(typeof(EndpointMetadataCollection))]
+#pragma warning restore RS0016 // Symbol 'implicit constructor for 'Enumerator'' is not part of the declared API.
```
Need for this looks like an analyzer bug. I seem to remember it's been fixed in more recent version of the package. But, switching versions is a larger change because it requires deeper changes in how we reference the package.

Pilchie replied:
Why does it need to be suppressed at all? Why not just add the constructor to the PublicAPI.Unshipped.txt?

dougbu commented:
Hold up, my branch is out-of-date. Need to re-up the baseline files

Pilchie commented on src/Http/Routing.Abstractions/src/PublicAPI.Unshipped.txt:
```diff
@@ -0,0 +1,118 @@
+Microsoft.AspNetCore.Http.EndpointMetadataCollection.EndpointMetadataCollection(System.Collections.Generic.IEnumerable<object!>! items) -> void (forwarded, contained in Microsoft.AspNetCore.Http.Abstractions)
+Microsoft.AspNetCore.Http.EndpointMetadataCollection.EndpointMetadataCollection(params object![]! items) -> void (forwarded, contained in Microsoft.AspNetCore.Http.Abstractions)
+Microsoft.AspNetCore.Http.EndpointMetadataCollection.Enumerator (forwarded, contained in Microsoft.AspNetCore.Http.Abstractions)
+Microsoft.AspNetCore.Http.EndpointMetadataCollection.Enumerator.Current.get -> object? (forwarded, contained in Microsoft.AspNetCore.Http.Abstractions)
```
Should be able to put that enumerator constructor right about here I think?

dougbu committed:
!fixup! Undo nit fixups

dougbu committed:
!fixup! Change baseline file instead of weird suppression
- revert suppression in an AssemblyInfo.cs file
- generally avoid manual edits to baseline files but just this once&hellip;

dougbu commented:
/azp run
